volumes/miovision/api/intersection_tmc.py

```diff
@@ -321,9 +321,19 @@ def process_data(conn, start_time, end_iteration_time):
     try:
         with conn:
             with conn.cursor() as cur:
-                update="SELECT miovision_api.aggregate_15_min_mvt(%s::date, %s::date)"
-                cur.execute(update, time_period)
-                logger.info('Aggregated to 15 minute movement bins')
+                #if intersections specified, aggregate this step one at a time
+                # with different single intersection function
+                if user_def_intersection:
+                    for c_intersec in intersections:
+                        update="""SELECT miovision_api.aggregate_15_min_mvt_single_intersection(
+                                    %s::date, %s::date, %s::int)"""
+                        cur.execute(update, time_period. c_intersec.uid)
```

Comment:
there's a period here instead of a comma. I also think you might need to expand the `time_period` tuple and pass one large tuple of parameters.